specification/hdinsight/data-plane/Microsoft.HDInsight/preview/2018-11-01-preview/job.json

```diff
@@ -0,0 +1,931 @@
+                    },
+                    {
+                        "name": "content",
+                        "in": "body",
+                        "required": true,
+                        "schema": {
+                            "type": "object",
+                            "format": "file"
+                        },
+                        "description": "The content of the Sqoop job request."
+                    }
+                ],
+                "responses": {
+                    "default": {
+                        "description": "Error response describing why the operation failed.",
+                        "schema": {
+                            "$ref": "#/definitions/JobOperationsErrorResponse"
+                        }
+                    },
+                    "200": {
+                        "description": "OK response definition.",
+                        "schema": {
+                            "$ref": "#/definitions/JobSubmissionJsonResponse"
+                        }
+                    }
+                }
+            }
+        },
+        "/ws/v1/cluster/apps/{appId}/state": {
+            "get": {
+                "tags": [
+                    "Job"
+                ],
+                "operationId": "Job_GetAppState",
+                "description": "Gets application state from the specified HDInsight cluster.",
+                "consumes": [
+                    "application/json"
+                ],
+                "produces": [
+                    "application/json"
+                ],
+                "x-ms-examples": {
+                    "Gets application state from the specified HDInsight cluster.": {
+                        "$ref": "./examples/HDI_Job_GetAppState.json"
+                    }
+                },
+                "parameters": [
+                    {
+                        "name": "appId",
+                        "in": "path",
+                        "required": true,
+                        "type": "string",
+                        "description": "The id of the job."
+                    }
+                ],
+                "responses": {
+                    "default": {
+                        "description": "Error response describing why the operation failed.",
+                        "schema": {
+                            "$ref": "#/definitions/JobOperationsErrorResponse"
+                        }
+                    },
+                    "200": {
+                        "description": "OK response definition.",
+                        "schema": {
+                            "$ref": "#/definitions/AppState"
+                        }
+                    }
+                }
+            }
+        }
+    },
+    "definitions": {
+        "AppState": {
+            "description": "The State of the application.",
+            "properties": {
+                "state": {
+                    "type": "string",

```

Comment:
If this is a well-defined set of values that will not change, it should be modeled as an enum.

Reply:
Good idea. Will update the state as an enum in the next iteration






---
In reply to: [237731666](https://github.com/Azure/azure-rest-api-specs/pull/4489#discussion_r237731666) [](ancestors = 237731666)